python/lsst/ap/association/packageAlerts.py

```diff
@@ -157,10 +223,13 @@ def run(self,
                                    objDiaForcedSources,
                                    diffImCutout,
                                    templateCutout))
-        with open(os.path.join(self.config.alertWriteLocation,
-                               f"{ccdVisitId}.avro"),
-                  "wb") as f:
-            self.alertSchema.store_alerts(f, alerts)
+
+        if self.config.doProduceAlerts and confluent_kafka is not None:
```

Comment:
If we `raise` in `__init__` for this case like I suggest above, the latter part of this is unnecessary.